#!/usr/bin/env python
PACKAGE = "yumi_motion"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# NAME		    TYPE	  RECONFIGURATION LEVEL	    DESCRIPTION               DEFAULT  	   MIN   MAX
gen.add("X_l",   	  double_t,      	 0,                  "x", 			0.46,      -1.0,   1.0)
gen.add("Y_l",      double_t,      	 0,                  "y", 			0.42,      -1.0,   1.0)
gen.add("L_l",      double_t,       	 0,                  "z", 			0.4,      -1.0,   1.0)
gen.add("Z_tcp_l",  double_t,       	 0,                  "z_tcp", 			0.2,      -1.0,   1.0)

gen.add("X_r",   	  double_t,      	 0,                  "x", 			0.46,      -1.0,   1.0)
gen.add("Y_r",      double_t,      	 0,                  "y", 			-0.42,      -1.0,   1.0)
gen.add("L_r",      double_t,       	 0,                  "z", 			0.4,      -1.0,   1.0)
gen.add("Z_tcp_r",  double_t,       	 0,                  "z_tcp", 			0.2,      -1.0,   1.0)

gen.add("n_solution",  int_t,       	 0,                  "n_solution", 		1,           1,   16)
gen.add("left",   bool_t,                0,                  "left", 					False)
gen.add("right",  bool_t,                0,                  "right", 					False)
gen.add("START",  bool_t,                0,                  "START", 					False)

gen.add("gripper_open_left",  bool_t,                0,                  "gripper_open_left", 					False)
gen.add("gripper_close_left",  bool_t,                0,                  "gripper_close_leftT", 					False)
gen.add("gripper_open_right",  bool_t,                0,                  "gripper_open_right", 					False)
gen.add("gripper_close_right",  bool_t,                0,                  "gripper_close_right", 					False)

 
exit(gen.generate(PACKAGE, "yumi_motion", "ik"))
